simple.js

```diff
@@ -0,0 +1,58 @@
+import * as helios from "@hyperionbt/helios"
+import fs from "fs";
+import {Color} from './colors.js'
+
+// 
+const contract = fs.readFileSync("simple.helios").toString();
```

Comment:
```javascript
✔ ~/workshop/potential-robot [test|✔] 
09:38 $ npm test

> potential-robot@1.0.0 test
> mocha --reporter list


Error: ENOENT: no such file or directory, open './simple.helios'
    at Object.openSync (node:fs:600:3)
    at Object.readFileSync (node:fs:468:35)
    at file:///home/alex/workshop/potential-robot/test/simple.js:5:21
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:438:15)
    at async formattedImport (/home/alex/workshop/potential-robot/node_modules/mocha/lib/nodejs/esm-utils.js:9:14)
    at async exports.requireOrImport (/home/alex/workshop/potential-robot/node_modules/mocha/lib/nodejs/esm-utils.js:42:28)
    at async exports.loadFilesAsync (/home/alex/workshop/potential-robot/node_modules/mocha/lib/nodejs/esm-utils.js:100:20)
    at async singleRun (/home/alex/workshop/potential-robot/node_modules/mocha/lib/cli/run-helpers.js:125:3)
    at async exports.handler (/home/alex/workshop/potential-robot/node_modules/mocha/lib/cli/run.js:370:5)
```

Reply:
Let me try Jest instead
https://nextjs.org/docs/testing
https://jestjs.io/
<details><summary>Details</summary>
<p>

```javascript
10:11 $ npm test                                                                                               [23/116]
                                                                                                                       
> potential-robot@1.0.0 test                                                                                           
> jest                                                                                                                 
                                                                                                                       
 FAIL  test/simple.test.js                                                                                             
  ● Test suite failed to run                                                                                           
                                                                                                                       
    Jest encountered an unexpected token                                                                               
                                                                                                                       
    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript synta
x, or when Jest is not configured to support such syntax.                                                              
                                                                                                                       
    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel co
nfiguration.                                                                                                           
                                                                                                                       
    By default "node_modules" folder is ignored by transformers.                                                       
                                                                                                                       
    Here's what you can do:                                                                                            
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript                
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in you
r config.                                                                                                              
     • If you need a custom transformation specify a "transform" option in your config.                                
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMa
pper" config option.
You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /home/alex/workshop/potential-robot/test/simple.test.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import * as helios from "@hyperio
nbt/helios";
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1495:14)

Test Suites: 1 failed, 1 total
``` 

</p>
</details>

Reply:
<details><summary>vitest</summary>
<p>

gav#4420

I've used vitest as a drop in replacement for JEST and Mocha.
https://github.com/vitest-dev/vitest/tree/main/examples/nextjs

```powershell
Serialized Error: {
  "code": "ENOENT",
  "errno": -2,
  "path": "simple.helios",
  "syscall": "open",
}

``` 

</p>
</details>

Reply:
`(cd test && node simple.test.js)`

[node](https://en.wikipedia.org/wiki/Node.js) is a [runtime](https://en.wikipedia.org/wiki/Runtime_system).

## execution
`simple.helios` is read as a string.
The `const` declaration creates block-scoped constants, much like variables declared using the [let](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) keyword.

[helios.Program.new](https://www.hyperion-bt.org/helios-book/api/reference/program.html?highlight=program#program)
program.compile -> [uplc](https://www.hyperion-bt.org/helios-book/api/reference/uplcprogram.html)

- [x] jest error, see below
- [x] https://vitest.dev/
- [x] comment:
https://github.com/aleeusgr/potential-robot/blob/3fa08443f8c966ce8f812a2321e5b5ea20603a80/test/simple.test.js#L15-L28

https://stackoverflow.com/questions/47982714/whats-the-difference-between-string-and-string-in-typescript

https://github.com/aleeusgr/potential-robot/issues/23